spark/src/main/scala/org/apache/spark/sql/delta/commands/DeleteWithDeletionVectorsHelper.scala

```diff
@@ -386,15 +388,36 @@ object DeletionVectorBitmapGenerator {
     DeletionVectorBitmapGenerator.buildDeletionVectors(sparkSession, df, txn.deltaLog, txn)
   }
 
-  private def buildCanonicalizedPathMap(
+  private def buildCanonicalizedPathStringMap(
       log: DeltaLog,
-      addFiles: Seq[AddFile]): Map[Path, String] = {
+      addFiles: Seq[AddFile]): Map[String, String] = {
     val basePath = log.dataPath.toString
     addFiles.map { add =>
       val absPath = absolutePath(basePath, add.path)
-      absPath -> SparkPath.fromPath(absPath).urlEncoded
+      absPath.toString -> SparkPath.fromPath(absPath).urlEncoded
     }.toMap
   }
+
+  /**
+   * In Spark 3.4 the file path metadata column is not canonicalized but it is in Spark 3.4.1. To
+   * make Delta Lake works with both Spark versions, we must use Spark's internal path
+   * transformation method to transform our paths here. This method will return a un-canonicalized
+   * path in Spark 3.4 and a canonicalized one in Spark 3.4.1.
+   *
+   * Related issue: https://github.com/delta-io/delta/issues/1725.
+   */
+  private lazy val sparkMetadataFilePathIsCanonicalized: Boolean = {
+    val probeString = "file:/path with space/data.parquet"
+    val row = FileFormat.updateMetadataInternalRow(
```

Comment:
Given this is not a public API, do we want to capture method not found error here and assume it is has the fix to return the path in URI encoded format?

Reply:
Good idea. Done!